### ISSUE TITLE
ODBC fails when using `TRIM(datname)`

### ISSUE THREAD

**Issue by bluesmoon** [JuliaDB/ODBC.jl#150]
With the following SQL statement run against a PostgreSQL or RedShift database:

```sql
SELECT TRIM(datname) FROM pg_database
```

I get the following error:
```
ERROR: MethodError: `copy!` has no method matching copy!(::Type{WeakRefString{UInt8}}, ::ODBC.Block{UInt8}, ::Array{Int64,1}, ::Array{WeakRefString{UInt8},1}, ::Int64, ::Int64)WARNING: Error showing method candidates, aborted

 in stream! at /home/ubuntu/.julia/v0.4/ODBC/src/backend.jl:167
 in stream! at /home/ubuntu/.julia/v0.4/ODBC/src/backend.jl:145
 in query at /home/ubuntu/.julia/v0.4/ODBC/src/backend.jl:256
```

This only happens when calling `TRIM` on `pg_database.datname`, so the following queries work:

```sql
SELECT TRIM(name) FROM stv_tbl_perm

SELECT datname FROM pg_database

```

Other info:
```
julia> Pkg.status("ODBC")
 - ODBC                          0.4.2

julia> VERSION
v"0.4.5"
```

**Comment by bluesmoon** [comment]
Some more info... in `backend.jl`, `Data.stream!`, 

```sql
SELECT datname FROM pg_database
```
has `rb.fetchsize === 5`

while
```sql
SELECT TRIM(datname) FROM pg_database
```
has `rb.fetchsize === 1`

**Comment by bluesmoon** [comment]
More info, it looks like `rb.indcols[col]` is of type `Array` instead of `NullableArray`, and this is what causes the problem.

**Comment by bluesmoon** [comment]
I tried changing it to a NullableArray, but that did not fix the issue.

**Comment by quinnj** [comment]
Hey @bluesmoon, have you had a chance to try and reproduce this on Julia 0.5 using ODBC 0.5.1?

**Comment by bluesmoon** [comment]
No, we stopped using ODBC.jl and instead I wrote a wrapper around PyCall to call a database specific driver depending on the database we use.  We are also unable to upgrade to Julia 0.5 at this point, and are unlikely to do so for at least another year as our customers are change averse.

**Comment by quinnj** [comment]
Understood. Plans are underway for Julia 1.0 to be released this summer, so that could be something to put on the radar as it will be the most feature-complete, long-support release for a while. Many packages are also already planning on targeting long-term support release to coincide.

**Comment by bluesmoon** [comment]
Thanks, that is good to know.  We might go right from 0.4 to 1.0.

**Comment by quinnj** [comment]
At this point, that's what I would recommend. 0.6 will be released imminently and 1.0 is next up, slated for this summer (maybe late summer before it gets all finalized).
